tox.ini

```diff
@@ -115,7 +115,7 @@ commands =
 [testenv:mypy]
 basepython = python3.4
 commands =
-    {[base]pip_install} mypy
+    {[base]pip_install} mypy==0.501
```

Comment:
I think part of the reason for the breakage was due to new, backwards-incompatible versions of mypy being released after #4386 landed. This pins the version to the one that was the latest as of the time of that PR's merging, [0.501](http://mypy-lang.blogspot.com/2017/03/mypy-0501-released.html).  (That said, we should definitely move to the latest version ASAP!)